TelemetryProcessors/PreAggregatedMetricsTelemetryProcessor.ts

```diff
@@ -0,0 +1,53 @@
+import Contracts = require("../Declarations/Contracts");
+import AutoCollecPreAggregatedMetrics = require("../AutoCollection/PreAggregatedMetrics");
+import * as TelemetryType from "../Declarations/Contracts";
+import {
+    MetricDependencyDimensions,
+    MetricExceptionDimensions,
+    MetricRequestDimensions,
+    MetricTraceDimensions
+} from "../Declarations/Metrics/AggregatedMetricDimensions";
+import Context = require("../Library/Context");
+
+export function preAggregatedMetricsTelemetryProcessor(envelope: Contracts.EnvelopeTelemetry, context: Context): boolean {
+    // Increment rate counters
+    switch (envelope.data.baseType) {
+        case TelemetryType.TelemetryTypeString.Exception:
+            let exceptionDimensions: MetricExceptionDimensions = {
+                cloudRoleInstance: envelope.tags[context.keys.cloudRoleInstance],
+                cloudRoleName: envelope.tags[context.keys.cloudRole],
+            };
+            AutoCollecPreAggregatedMetrics.countException(exceptionDimensions);
+            break;
+        case TelemetryType.TelemetryTypeString.Trace:
+            let traceDimensions: MetricTraceDimensions = {
+                cloudRoleInstance: envelope.tags[context.keys.cloudRoleInstance],
+                cloudRoleName: envelope.tags[context.keys.cloudRole],
```

Comment:
`trace/count` requires dimensions `trace/severityLevel` and  `operation/synthetic` as well.